docs/mpcie_modules.md

```diff
@@ -162,6 +162,50 @@ operating system.
 
 (1) Refer to quirks, it is known to not work stably on all apu2 board revisions
 
+## Ethernet controllers
+
+### Dual Ethernet Controller Realtek RTL8111
+
+![RTL8111](images/realtek_controller.jpg)
+
+This model works well with apu1 boards. However, it is not seen in apu2 and apu3
+boards with default coreboot settings. To enable it make following steps:
+
+1. On apu2 board make sure to enable option *force mPCIe2 clk* in Payload menu.
+This menu is available during boot process by pressing `F10` button. On apu3 it
+doesn't affect detection by OS. When above option is enabled, you should see
+RTL8111 as PCI device in your OS.
+
+2. Manually download and install missing Realtek firmware. Unfortunately,
+coreboot doesn't support it's firmware during boot process. Therefor, even if
+Ethernet controller is recognized it won't work. Installing missing firmware can
+be done, for example in Debian OS, by performing those steps:
+
+    - Edit `/etc/apt/sources.list` file. It should contain additional two
+    lines:
+    ```
+    deb http://ftp.pl.debian.org/debian stretch main non-free
+    deb-src  http://ftp.pl.debian.org/debian stretch main non-free   
+    ```
+    - Update apt-get:
+    ```
+    sudo apt-get update
+    ```
+    - Install Realtek firmware
+    ```
+    sudo apt-get install firmware-realtek
+    ```
+
+Now RTL8111 controller should works fine and 2 additional Ethernet Network
+devices should appear.
+
+**Compatible slots**
+
+| Platform |  mPCIe1  |     mPCIe2      |  mSATA   |
+|:--------:|:--------:|:---------------:|:--------:|
+|   apu2   | &#10004; | &#10004;        | &#10006; |
+|   apu3   | &#10004; | &#10006;        | &#10006; |
+
```

Comment:
@piotr-kleins testing on apu4 would be great too

Reply:
apu4 tested